langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiEmbeddingModel.java

```diff
@@ -122,8 +130,16 @@ public Response<List<Embedding>> embedAll(List<TextSegment> segments) {
 
                 List<Value> instances = new ArrayList<>();
                 for (TextSegment segment : batch) {
+                    VertexAiEmbeddingInstance embeddingInstance = new VertexAiEmbeddingInstance(segment.text());
+                    // Title metadata is used for calculating embeddings for document retrieval
+                    embeddingInstance.setTitle(segment.metadata("title"));
```

Comment:
[Docu](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings) says title is valid only for `RETRIEVAL_DOCUMENT`.
Would be nice to document this in hte javadoc of this class.
I would also give user an option to define the key of metadata to use instead of hardcoding "title".

Reply:
How would you let the user customize the `title` property?

Reply:
Perhaps something like `titleMetadataKey("my_title")`, but this is not urgent, can be added later. My main concern was in that if there is a "title" metadata entry and task type is not `RETRIEVAL_DOCUMENT`, then embedding call might fail?

Reply:
Indeed that's a bit weird, but that's what's happening: using `title` without `RETRIEVAL_DOCUMENT` would fail.
I'll add the suggested method to customize the metadata key.

Reply:
Pushed the extra builder method.

Reply:
Maybe worth adding a condition here? Set `title` only for `RETRIEVAL_DOCUMENT` ?

Reply:
Good point: pushed a change to set `title` only for `RETRIEVAL_DOCUMENT`.
That way, it's still possible to use `title` when not using that embedding task type.